tests/manage/mcg/test_namespace_crd.py

```diff
@@ -784,6 +790,148 @@ def test_create_ns_bucket_from_utilized_resources_crd(
+        ).get()
+        assert (
+            "has buckets attached" in conditions["status"]["conditions"][0]["message"]
+        )
+
+    @pytest.mark.polarion_id("OCS-2283")
+    @tier3
+    def test_create_bucket_with_invalid_namespacestore_crd(
+        self, namespace_store_factory, bucket_factory
+    ):
+        """
+        Test that a proper error message is reported when invalid
+        NamespaceStore name is provided during namespace bucket creation.
+        """
+        logger.info("Create one valid namespace resource and verify health")
+        nss_tup = ("oc", {"aws": [(1, self.DEFAULT_REGION)]})
+        valid_store = namespace_store_factory(*nss_tup)[0]
+
+        class InvalidNamespacestore:
+            """
+            Empty class that contains invalid NamespaceStore name
+            """
+
+            def __init__(self, name):
+                self.name = name
+
+        for store_name in ("", "nonexistent"):
+            logger.info(
+                f"Trying to create namespace bucket with write resource set to '{store_name}'"
+            )
+            bucketclass_dict = {
+                "interface": "OC",
+                "namespace_policy_dict": {
+                    "type": "Multi",
+                    "namespacestores": [InvalidNamespacestore(store_name), valid_store],
+                },
+            }
+            bucket = bucket_factory(
+                amount=1,
+                interface=bucketclass_dict["interface"],
+                bucketclass=bucketclass_dict,
+                verify_health=False,
+            )[0]
+            try:
+                bucket.verify_health(timeout=15)
+            except UnhealthyBucket:
+                logger.info(
+                    "Bucket with invalid NamespaceStore is not healthy as Expected"
+                )
+            else:
+                assert False, "Bucket with invalid NamespaceStore should not be healthy"
+            bucket.delete()
+
+            logger.info(
+                f"trying to create namespace bucket with read resource set to '{store_name}'"
+            )
+            bucketclass_dict = {
+                "interface": "OC",
+                "namespace_policy_dict": {
+                    "type": "Multi",
+                    "namespacestores": [valid_store, InvalidNamespacestore(store_name)],
+                },
+            }
+            bucket = bucket_factory(
+                amount=1,
+                interface=bucketclass_dict["interface"],
+                bucketclass=bucketclass_dict,
+                verify_health=False,
+            )[0]
+            try:
+                bucket.verify_health(timeout=15)
+            except UnhealthyBucket:
+                logger.info(
+                    "Bucket with invalid NamespaceStore is not healthy as Expected"
+                )
+            else:
+                assert False, "Bucket with invalid NamespaceStore should not be healthy"
+            bucket.delete()
```

Comment:
Same as 893 to 907 -
```suggestion
            try:
                bucket = bucket_factory(bucketclass=bucketclass_dict)[0]
            except UnhealthyBucket:
                logger.info(
                    "Bucket with invalid NamespaceStore is not healthy as Expected"
                )
            else:
                assert False, "Bucket with invalid NamespaceStore should not be healthy"
```